strawberry/annotation.py

```diff
@@ -269,12 +275,19 @@ def _strip_lazy_type(cls, annotation: LazyType) -> type:
 
 
 def _is_input_type(type_: Any) -> bool:
-    if not _is_object_type(type_):
+    if not _get_strawberry_definition(type_):
         return False
 
-    return type_._type_definition.is_input
+    return type_.__strawberry_definition__.is_input
 
 
-def _is_object_type(type_: Any) -> bool:
-    # isinstance(type_, StrawberryObjectType)  # noqa: E800
-    return hasattr(type_, "_type_definition")
+def _get_strawberry_definition(type_: Any) -> Optional[StrawberryDefinition]:
+    origin = type_
+    # generics store their class in __origin__
+    if origin_ := getattr(type_, "__origin__", False):
+        origin = origin_
+    res = getattr(origin, "__strawberry_definition__", None)
+    if isinstance(res, StrawberryDefinition):
+        return res
+    else:
+        return None
```

Comment:
You implemented this function (which I think is great!), but there are a lot of places where we're still manually grabbing the `__strawberry_definition__` off the types. I think it should be:
1. be moved out of `annotations.py`
2. used in more places

Reply:
Where do you think it should live?